### PR TITLE
WIP complete tab animation

### PR DESCRIPTION
**Pull request by chris-saburo**
#143 
- completed Tab animation

### CODE REVIEW

File: src/views/ArticleLists.vue [hunk 1]
```diff
@@ -31,20 +46,21 @@
 
 <script>
 import BaseAppBarHeader from '@/components/BaseAppBarHeader.vue';
-import BaseTabs from '@/components/BaseTabs.vue';
 import BaseTab from '@/components/BaseTab.vue';
 import ArticleItem from '@/components/ArticleListsItem.vue';
 import LayoutFixedScrollable from '@/components/LayoutFixedScrollable.vue';
 import BaseNavigationBottom from '@/components/BaseNavigationBottom.vue';
+import Tabs from '@/components/BaseTabsAnimation.vue';
```
Comment (mort3za) [inline src/views/ArticleLists.vue:53]: Please don't rename components on import.

File: src/components/BaseTabsAnimation.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
+<template>
```
Comment (mort3za) [inline src/components/BaseTabsAnimation.vue:1]: Do we need a separate component for animated tabs? Why not using the same BaseTabs?  
I think adding animation should be an option in BaseTabs with ability of enable/disable it, unless there is a good reason to separate them.

File: src/helpers.js [hunk 3]
```diff
@@ -161,6 +161,19 @@ export async function sleep(time) {
   });
 }
 
+export function getTabMeta() {
```
Comment (mort3za) [inline src/helpers.js:164]: You are using this function only in ArticleLists, so you don't need to define a helper.

File: src/styles/animations.css [hunk 4]
```diff
@@ -8,3 +8,53 @@
...
+  position: relative;
+  margin: 0 auto;
+  &__item {
+    display: inline-block;
+    margin: 0 5px;
+    padding: 10px;
+    padding-bottom: 8px;
+    font-size: 16px;
+    letter-spacing: 0.8px;
+    color: gray;
```
Comment (mort3za) [inline src/styles/animations.css:21]: No hardcoded colors :)

File: src/styles/animations.css [hunk 5]
```diff
@@ -8,3 +8,53 @@
...
+    &:focus {
+      outline: none;
+      border-bottom: 2px solid gray;
+      color: blue;
+    }
+    &_active {
+      color: blue;
+    }
+    &:disabled {
+      border-bottom: 2px solid transparent;
```
Comment (mort3za) [inline src/styles/animations.css:41]: We can squash most of these styles by using tailwind's @apply. We have these styles in tailwind.

File: src/helpers.js (hunk 3 above)
Comment (chris-saburo) [reply src/helpers.js:164]: Done it.

File: src/styles/animations.css (hunk 4 above)
Comment (chris-saburo) [reply src/styles/animations.css]: Done it.

File: src/styles/animations.css (hunk 5 above)
Comment (chris-saburo) [reply src/styles/animations.css]: I think it needs to use & way in this tab animation case.

File: src/components/BaseTabsAnimation.vue (hunk 2 above)
Comment (chris-saburo) [reply src/components/BaseTabsAnimation.vue:1]: I don't think so.

File: src/views/ArticleLists.vue (hunk 1 above)
Comment (chris-saburo) [reply src/views/ArticleLists.vue]: Done it.

File: src/components/BaseTabsAnimation.vue (hunk 2 above)
Comment (mort3za) [reply src/components/BaseTabsAnimation.vue:1]: Explain please

File: src/styles/animations.css (hunk 5 above)
Comment (mort3za) [reply src/styles/animations.css]: `&` is fine, apply has no problem with it.

**Pull request closed without merging**